### PR TITLE
Add an index that can be used to check if a field is set or not

### PR DESCRIPTION
**Pull request by dadgar**
Add an index that can be used to check if a field is set or not by checking if the field has the types nil value.

### CODE REVIEW

**Comment by dadgar** [comment]
@armon: Could you review

File: index.go [hunk 1]
```diff
@@ -175,6 +175,44 @@ func (u *UUIDFieldIndex) parseString(s string) ([]byte, error) {
...
+// FieldSetIndex is used to extract a field from an object using reflection and
+// builds an index on whether the field is set by comparing it against its
+// type's nil value.
+type FieldSetIndex struct {
+	Field string
+}
+
+func (f *FieldSetIndex) FromObject(obj interface{}) (bool, []byte, error) {
+	v := reflect.ValueOf(obj)
+	v = reflect.Indirect(v) // Derefence the pointer if any
```
Comment (slackpad) [inline index.go]: "Dereference"

File: index_test.go [hunk 2]
```diff
@@ -225,6 +228,94 @@ func generateUUID() ([]byte, string) {
 	return buf, uuid
 }
 
+func TestFieldSetIndex_FromObject(t *testing.T) {
+	obj := testObj()
+	ptrIndexer := FieldSetIndex{"Bam"}
```
Comment (slackpad) [inline index_test.go]: Why name it `ptrIndexer`? It's not a pointer so I'd just call it `indexer`.

File: integ_test.go [hunk 3]
```diff
@@ -184,10 +217,18 @@ func testPopulateData(t *testing.T, db *MemDB) {
 
 	// Create some data
 	person1 := testPerson()
+
 	person2 := testPerson()
 	person2.First = "Mitchell"
 	person2.Last = "Hashimoto"
 
+	person3 := testPerson()
```
Comment (slackpad) [inline integ_test.go]: Nice!

File: integ_test.go [hunk 4]
```diff
@@ -184,10 +217,18 @@ func testPopulateData(t *testing.T, db *MemDB) {
...
+	person3.First = "Alex"
+	person3.Last = "Dadgar"
+
+	person1.Sibling = person3
```
Comment (slackpad) [inline integ_test.go]: Nice!

**Comment by slackpad** [comment]
Just two tiny nits (typo and a name change), otherwise LGTM.

**Pull request merged**